staging/src/k8s.io/kubectl/pkg/explain/v2/generator.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"fmt"
+	"io"
+	"text/template"
+
+	sprig "github.com/go-task/slim-sprig"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type TemplateContext struct {
+	GVR       schema.GroupVersionResource
+	Document  map[string]interface{}
+	Recursive bool
+	FieldPath []string
+}
+
+type generator struct {
+	document  map[string]interface{}
+	templates map[string]*template.Template
+}
+
+func NewGenerator(
```

Comment:
It should be easy to write a "simple" unit-test for this code?

Add a dummy template that prints the context, render it with an arbitrary simple context, test that renders gives the right thing?

Reply:
added unit tests